src/hotspot/share/gc/shenandoah/shenandoahFullGC.cpp

```diff
@@ -823,11 +829,13 @@ class ShenandoahEnsureHeapActiveClosure: public ShenandoahHeapRegionClosure {
 public:
   ShenandoahEnsureHeapActiveClosure() : _heap(ShenandoahHeap::heap()) {}
   void heap_region_do(ShenandoahHeapRegion* r) {
+    bool is_generational = _heap->mode()->is_generational();
     if (r->is_trash()) {
       r->recycle();
     }
     if (r->is_cset()) {
-      r->make_regular_bypass();
+      // If generational, leave affiliation alone.
+      r->make_regular_bypass(!is_generational);
```

Comment:
So - if we are not in generational mode, we will set the affiliation to `YOUNG`, is that necessary when we are not in generational mode? Wouldn't it already be young (or free) in that mode?

And when we are in generational mode, we don't want to change the affiliation. Should we just remove this affiliation changing side-effect of `make_regular_bypass`?

Reply:
You raise good questions here.  I was not sure why make_regular_bypass() wanted to overwrite affiliation with young.  I knew it shouldn't do it in the case I was observing.  I was trying to effect only my use case so as not to break other "use cases".

I'll think a bit harder on this.